python/spacewalk/satellite_tools/spacewalk-repo-sync

```diff
@@ -65,100 +66,226 @@ def main():
+        help="Comma or space separated list of excluded packages or package groups.",
+    )
+    parser.add_option(
+        "",
+        "--no-strict",
+        action="store_true",
+        help="do not unlink packages when deleted from repository",
+        dest="no_strict",
+    )
+    parser.add_option(
+        "",
+        "--email",
+        action="store_true",
+        help="e-mail a report of what was synced/imported",
+    )
+    parser.add_option(
+        "",
+        "--traceback-mail",
+        action="store",
+        help="alternative email address(es) for sync output (--email option)",
+    )
+    parser.add_option(
+        "",
+        "--no-errata",
+        action="store_true",
+        dest="no_errata",
+        default=False,
+        help="Do not sync errata",
+    )
+    parser.add_option(
+        "",
+        "--no-packages",
+        action="store_true",
+        dest="no_packages",
+        default=False,
+        help="Do not sync packages",
+    )
+    parser.add_option(
+        "",
+        "--sync-kickstart",
+        action="store_true",
+        dest="sync_kickstart",
+        default=False,
+        help="Sync kickstartable tree",
+    )
+    parser.add_option(
+        "",
+        "--force-all-errata",
+        action="store_true",
+        dest="force_all_errata",
+        default=False,
+        help="Process metadata of all errata, not only missing.",
+    )
+    parser.add_option(
+        "",
+        "--batch-size",
+        action="store",
+        help="max. batch size for package import (debug only)",
+    )
+    parser.add_option(
+        "-Y",
+        "--deep-verify",
+        action="store_true",
+        dest="deep_verify",
+        default=False,
+        help="Do not use cached package checksums",
+    )
+    parser.add_option(
+        "-v",
+        "--verbose",
+        action="count",
+        help="Verbose output. Possible to accumulate: -vvv",
+    )
+    (options, _) = parser.parse_args()
 
     global LOCK
     try:
-        LOCK = rhnLockfile.Lockfile('/run/spacewalk-repo-sync.pid')
+        LOCK = rhnLockfile.Lockfile("/run/spacewalk-repo-sync.pid")
     except rhnLockfile.LockfileLockedException:
-        systemExit(1, "ERROR: attempting to run more than one instance of "
-                      "spacewalk-repo-sync Exiting.")
+        systemExit(
+            1,
+            "ERROR: attempting to run more than one instance of "
+            "spacewalk-repo-sync Exiting.",
+        )
 
     log_level = options.verbose
     if log_level is None:
-        log_level = 0
-
-    initCFG('server.satellite')
-    CFG.set('DEBUG', log_level)
+        # if no -v flag is passed, use "debug" setting from rhn.conf
+        with cfg_component(None) as cfg:
+            log_level = cfg.get("DEBUG", 0)
+
+    # Apparently we need to call initCFG and have CFG available
+    # to prevent some errors accessing CFG later on during package
+    # import.
+    initCFG("server.satellite")
```

Comment:
Can we please fix this as well? I'm okay with doing it in another PR, as we already have the patch we shouldn't leave it lying around until it does not apply anymore.

Reply:
I'd vote for doing a separated PR.

In our initial attempt, we noticed that other python files that are part of the reposync execution stack were relying on CFG being preinitializated to `server.satellite`, and that causes some issues in certain corner cases. Instead of going forwad with the refactor, in the context of the bug fix, we just isolated the actual logging improvement from the refactor to use context manager (that was causing some issues). I agree we can just fix them, but we didn't by then.

I would say we can prepare another PR with the refactor, plus any extra fixes required to use context managers, just after this one.